File: Ruby/palindrome.rb
# Without regex

def palindrome(text)
  text = text.downcase
  text_letters = text.split('')
  all_chars = 'abcdefghijklmnopqrstuvwxyz'.split('')

  letters_array = []
  text_letters.map do |char|
    letters_array.push(char) if all_chars.include?(char)
  end
  letters_array.join('') == letters_array.reverse.join('') ? true : false
end

puts palindrome("Madam I'm Adam")

File: JavaScript/palindrome.js
// Not using Regexp

function isPalindrome(text) {
  text = text.toLowerCase()
  const textChars = text.split('')
  const allChars = 'abcdfghijklmnopqrstuvwxyz'.split('')

  const letterArray = []
  textChars.map( c => {
    if ( allChars.includes(c) ) letterArray.push(c)
  })
  if ( letterArray.join('') === letterArray.reverse().join('') ) return true
  else return false
}
console.log(isPalindrome("Madam I'm Adam"))